File: uci-user-interface/src/app/interfaces/list-transportational-request.ts
import { ProfileResponse } from "./profile-response"

export interface ListTransportationalRequest {
    _id: string,
    product: {
        "_id": string,
        "product_name": string,
        "owner": ProfileResponse,
        "description": string,
        "price": number,
        "quantity": number,
        "product_image": string,
        "transportation_type": string
    },
    user: ProfileResponse,
    quantity: number,
    status: string,
    posted_at: string

}

File: unified-commodities-interface/src/main/java/com/example/unifiedcommoditiesinterface/models/dto/ListTransportationalRequest.java
package com.example.unifiedcommoditiesinterface.models.dto;

import com.example.unifiedcommoditiesinterface.models.Profile;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class ListTransportationalRequest {
    private String _id;
    
    private ProductListing product;
    
    private Profile user;

    private Double quantity;
    
    private String status;

    private String posted_at;
}